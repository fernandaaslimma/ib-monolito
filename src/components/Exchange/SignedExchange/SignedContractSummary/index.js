import SignedContractSummary from "./SignedContractSummary";

export default SignedContractSummary;
