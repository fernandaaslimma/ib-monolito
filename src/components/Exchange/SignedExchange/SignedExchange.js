import React, { Component } from "react";
import { arrayOf, shape, func } from "prop-types";

import { Container } from "../../../styles/grid";
import { translate } from "../../../utils/i18n";
import { ExchangeWrapper } from "./styles";
import SignedContractSummary from "./SignedContractSummary";
import DefaultContent from "../../common/DefaultContent";
import Icon from "../../common/Icon";
import { black30 } from "../../../styles/settings";

import ErrorBoundary from "../../ErrorBoundary/ErrorBoundary";

class SignedExchange extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isEmpty: false,
      loading: true
    };
  }

  async componentWillMount() {
    await this.props.getSignedContracts();
    this.setState({
      isEmpty: this.props.signedContracts.length === 0,
      loading: false
    });
  }

  renderContent(contracts, isEmpty, loading) {
    if (!isEmpty) {
      return contracts.map(contract => (
        <SignedContractSummary
          key={contract.contractId}
          contract={contract}
          data-test={`exchangeContract${contract.contractId}`}
          loading={loading}
          {...contract}
        />
      ));
    }
    return (
      <DefaultContent
        Icon={() => (
          <Icon width="140" height="114" type="Papers" color={black30} />
        )}
        data-test="NoContracts"
        primaryText={translate("NO_CONTRACTS")}
        secondaryTexts={[translate("THERE_ARE_NO_SIGNED_CONTRACTS")]}
      />
    );
  }

  render() {
    const { signedContracts, error } = this.props;
    const { isEmpty, loading } = this.state;

    return (
      <ErrorBoundary errorStatus={error}>
        <ExchangeWrapper>
          <Container data-test="SignedExchange">
            {this.renderContent(signedContracts, isEmpty, loading)}
          </Container>
        </ExchangeWrapper>
      </ErrorBoundary>
    );
  }
}

SignedExchange.defaultProps = {
  signedContracts: []
};

SignedExchange.propTypes = {
  signedContracts: arrayOf(shape(SignedContractSummary.propTypes)),
  getSignedContracts: func.isRequired // eslint-disable-line
};

export default SignedExchange;
