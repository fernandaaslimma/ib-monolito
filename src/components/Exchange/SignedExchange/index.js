import { connect } from "redux-zero/react";

import SignedExchange from "./SignedExchange";
import contractsActions from "../../../actions/contracts";

export default connect(
  ({ signedContracts, loading, error }) => ({
    signedContracts,
    loading,
    error
  }),
  contractsActions
)(SignedExchange);
