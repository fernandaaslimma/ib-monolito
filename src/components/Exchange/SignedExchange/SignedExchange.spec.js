import React from "react";
import { shallow } from "enzyme";

import SignedExchange from "./SignedExchange";

jest.mock("../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

describe("SignedExchange component", () => {
  let getSignedContractsMock;

  beforeEach(() => {
    getSignedContractsMock = jest.fn();
  });

  it("should match snapshot", () => {
    expect(
      shallow(<SignedExchange getSignedContracts={getSignedContractsMock} />)
    ).toMatchSnapshot();
  });

  it("should trigger Contract getSignedContracts once when mounting", () => {
    shallow(<SignedExchange getSignedContracts={getSignedContractsMock} />);

    expect(getSignedContractsMock).toHaveBeenCalledTimes(1);
  });

  it("should render contract list if signedContracts Object is fullfiled", () => {
    const component = shallow(
      <SignedExchange
        signedContracts={[{ contractId: "otherthing" }]}
        getSignedContracts={getSignedContractsMock}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
