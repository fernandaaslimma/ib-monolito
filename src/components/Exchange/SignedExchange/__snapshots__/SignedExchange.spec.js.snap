// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedExchange component should match snapshot 1`] = `
<ErrorBoundary>
  <styled.div>
    <Container
      data-test="SignedExchange"
    />
  </styled.div>
</ErrorBoundary>
`;

exports[`SignedExchange component should render contract list if signedContracts Object is fullfiled 1`] = `
<ErrorBoundary>
  <styled.div>
    <Container
      data-test="SignedExchange"
    >
      <SignedContractSummary
        contract={
          Object {
            "contractId": "otherthing",
          }
        }
        contractId="otherthing"
        data-test="exchangeContractotherthing"
        key="otherthing"
        loading={true}
      />
    </Container>
  </styled.div>
</ErrorBoundary>
`;
